### PR TITLE
size, index, cat w/o dimension yet

### PR DESCRIPTION
**Pull request by JernKunpittaya**
Implement Size, Gather, Concat in a limited way that supports
-x.size()[0], x.size()[1], etc.
-Getting a certain number out of array i.e. X[0][0]
-Concat two tensor horizontally: [1,3,4] concat [4,3,2] --> [1,3,4,4,3,2]

This doesn't cover edge case like concat vertically (torch.cat(dim = 1) yet). Or to make circom template that realizes that Given 2D X, X[0] should be 1D array. Now we just treat X[0] as just one value since we know that all our tensors are in shape (-1,1) anyway. This should work fine for all zk-stats supported operations but can cause problem for Regression, which has something like torch.cat((*args[:-1], torch.ones_like(args[0])), dim = 1)

I think to really make circom template truly reflect the shape of keras tensor, we need to change every of our template now , for example in mpc.circom, rn TFEqual, TFGreater, etc. just considers the signal input as 1D instead of 2D. If we really want to make array indexing X[0] really represents 1D array, we need to adjust those templates as well. 

My suggestion is to try implement every function with this dimension-ignorant approach first, and we can see what's the best way to make it work for Regression later on. lemme know what u think @mhchia

### CODE REVIEW

**Comment by JernKunpittaya** [comment]
We agreed to proceed as suggested :)
